app/src/components/common/big_number_input/index.tsx

```diff
@@ -25,6 +25,7 @@ interface Props {
 
 interface State {
   currentValueStr: string
+  currentDecimalsStr: number
```

Comment:
I don't get this. Before `this.props.decimals` was being used. Now there's this state variable that's only updated when `this.props.decimals` change. It seems to me like it should have the same behavior as before. What am I missing here?

Reply:
If we have tokens with different decimals, the component will throws an exception

Reply:
I'm talking about something internal to the component. Using a prop directly, or setting a state when the prop changes and then using that state, should be equivalent (but the later adds unnecessary complexity). If using the prop directly throws an error, then we should investigate why that happen, because it shouldn't.